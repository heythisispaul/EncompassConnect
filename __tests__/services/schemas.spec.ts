import { testInstanceWithToken } from '../__utils__/mockEncompassConnectInstances';
import { mockResponse, fetch } from '../__utils__/mockFunctions';
import { matchesFetchSnapshot, returnsResponseBody } from '../__utils__/commonTests';

jest.mock('node-fetch');

describe('schemas', () => {
  beforeEach(() => {
    fetch.mockReset();
  });
  describe('generateContract', () => {
    const mockValues = {
      317: 'Todd Chavez',
      2: '10000',
    };

    it('makes the API call with the correct information', async () => {
      mockResponse();
      await testInstanceWithToken.schemas.generateContract(mockValues);
      expect(fetch.mock.calls[0]).toMatchSnapshot();
    });

    it('returns the body of the response', async () => (
      returnsResponseBody('schemas.generateContract', mockValues)
    ));
  });

  describe('getLoanSchema', () => {
    it('makes the API call with the correct information', async () => (
      matchesFetchSnapshot('schemas.getLoanSchema')
    ));

    it('returns the body of the response', async () => (
      returnsResponseBody('schemas.getLoanSchema')
    ));

    it('adds the entities from the provided array to the query string', async () => {
      const entities = ['princess', 'caroline'];
      mockResponse();
      await testInstanceWithToken.schemas.getLoanSchema(entities);
      // @ts-ignore
      const path: string = fetch.mock.calls[0][0];
      expect(path.includes('?entities=princess,caroline')).toBeTruthy();
    });
  });
});
