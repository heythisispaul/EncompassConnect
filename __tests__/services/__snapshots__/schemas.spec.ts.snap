// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schemas generateContract makes the API call with the correct information 1`] = `
Array [
  "https://api.elliemae.com/encompass/v1/schema/loan/contractGenerator",
  Object {
    "body": "{\\"2\\":\\"10000\\",\\"317\\":\\"Todd Chavez\\"}",
    "headers": Object {
      "Authorization": "Bearer <MOCK TOKEN VALUE>",
    },
    "method": "POST",
  },
]
`;

exports[`schemas getLoanSchema makes the API call with the correct information 1`] = `
Array [
  "https://api.elliemae.com/encompass/v1/schema/loan",
  Object {
    "headers": Object {
      "Authorization": "Bearer <MOCK TOKEN VALUE>",
    },
  },
]
`;
